game/src/core/components/DrawComponent.java

```diff
@@ -47,35 +47,35 @@
  * @see Animation
  * @see IPath
  */
-public class DrawComponent extends Component {
+public final class DrawComponent extends Component {
     private final Map<String, Animation> animationMap;
     private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
     private Animation currentAnimation;
 
     /**
-     * Create a new DrawComponent.
+     * Create a new DrawComponent and add it to the associated entity..
      *
      * <p>Will read in all subdirectories of the given path and use each file in the subdirectory to
      * create an animation. So each subdirectory should contain only the files for one animation.
      *
      * <p>Will set the current animation to idle left
      *
      * @param entity associated entity
-     * @param path Path (as a string) to the directory in the assets folder where the subdirectories
-     *     containing the animation files are stored. Example: "character/knight".
+     * @param path Path (as a string) to the directory in the assets folder where the
+     *     subdirectories containing the animation files are stored. Example: "character/knight".
      * @throws IOException if the given path does not exist
      * @see Animation
      */
-    public DrawComponent(Entity entity, String path) throws IOException {
+    public DrawComponent(final Entity entity, final String path) throws IOException {
         super(entity);
         // fetch available animations
         ClassLoader classLoader = getClass().getClassLoader();
-        File directory = new File(classLoader.getResource(path).getFile());
+        File directory = new File(Objects.requireNonNull(classLoader.getResource(path)).getFile());
         if (!directory.exists() || !directory.isDirectory()) {
             throw new FileNotFoundException("Path " + path + " not found.");
         }
         animationMap =
-                Arrays.stream(directory.listFiles())
+                Arrays.stream(Objects.requireNonNull(directory.listFiles()))
```

Comment:
Hier ist es aber wirklich ganz falsch. Schau mal: https://docs.oracle.com/javase/8/docs/api/java/io/File.html#listFiles-- :

"If this abstract pathname does not denote a directory, then this method returns null. Otherwise an array of File objects is returned ..." 

Das Ding ist nur dann `null`, wenn das `directory` kein Verzeichnis ist. Und das haben wir exakt etwas weiter oben abgefragt.

Reply:
Ich frage mich, ob IJ bereits mit ChatGPT arbeitet. Wirkt grad so auf mich 💩🙈

Reply:
Du und dein ChatGPT mobbing.
Warte mal bis sich die KIs den großen Aufstand planen... :P

Reply:
hihi ... dann stehe ich vermutlich in der ersten reihe (vor der wand) ...

Reply:
aber ist doch so. ich finde es ja nett, wenn ij ne menge vorschläge hat. aber in letzter zeit scheint da murks vor qualität zu gehen und ich kann mir das irgendwie kaum noch anders erklären, als dass die einfach ihr regelbasiertes system gegen irgendwas selbstlernendes ausgetauscht haben ... krieg's ja auch mit, was da so kommt.

Reply:
> und wenn du dann anwender hast, die das nicht hinterfragen oder vielleicht gar nicht erst einordnen können, dann wird es richtig blöd.

ich fass mir mal sanft selbst an die Nase

Reply:
> > und wenn du dann anwender hast, die das nicht hinterfragen oder vielleicht gar nicht erst einordnen können, dann wird es richtig blöd.
> 
> 
> 
> ich fass mir mal sanft selbst an die Nase

dachte eigentlich eher an die studis 🤣